lib/rack/query_parser.rb

```diff
@@ -86,6 +86,7 @@ def normalize_params(params, name, v, depth)
       name =~ %r(\A[\[\]]*([^\[\]]+)\]*)
```

Comment:
Do you think you can tidy up this parsing logic a bit? Using `$` variables concerns me a little.

Reply:
I can, but I don't see the reason to do so in this PR.  It's better to keep the implementation the same as before.  After all, there are other `$` variables used elsewhere in the code, including later in this method.  If you would like to remove `$` variable usage, we should probably do that globally in a separate PR.